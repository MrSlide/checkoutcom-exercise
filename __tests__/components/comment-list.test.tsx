import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'

import Comment from '../../src/components/comment'
import CommentList from '../../src/components/comment-list'

describe('CommentList component', function () {
  const testCommentProps = [
    {
      author: 'John Doe',
      comment: 'Lorem ipsum dolore sit amet.',
    },
    {
      author: 'Jane Doe',
      comment: 'Duis finibus pretium mi, eu maximus quam mollis ac.',
    }
  ]

  test('renders correctly with comments', function () {
    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <CommentList>
        <Comment {...testCommentProps[0]}/>
        <Comment {...testCommentProps[1]}/>
      </CommentList>
    )

    expect(result).toMatchSnapshot()
  })

  test('renders correctly without comments', function () {
    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <CommentList/>
    )

    expect(result).toMatchSnapshot()
  })
})
