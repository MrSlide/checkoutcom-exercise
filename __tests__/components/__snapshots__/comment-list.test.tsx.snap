// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList component renders correctly with comments 1`] = `
<ol
  className="comments-list"
>
  <li
    className="comments-list__item"
  >
    <Comment
      author="John Doe"
      comment="Lorem ipsum dolore sit amet."
    />
  </li>
  <li
    className="comments-list__item"
  >
    <Comment
      author="Jane Doe"
      comment="Duis finibus pretium mi, eu maximus quam mollis ac."
    />
  </li>
</ol>
`;

exports[`CommentList component renders correctly without comments 1`] = `
<Paragraph>
  There are no comments yet.
</Paragraph>
`;
